drivers/gpu/drm/i915/i915_active.c

```diff
@@ -130,6 +131,55 @@ static inline void debug_active_assert(struct i915_active *ref) { }
 
 #endif
 
+#if defined(__FreeBSD__)
+static struct llist_head linux_kmem_cache_list = LLIST_HEAD_INIT();
+
+struct cache_addr {
+	struct linux_kmem_cache *cache;
+	struct llist_node *next;
+};
+
+static void
+linux_kmem_cache_free_async_fn(struct irq_work *irqw)
+{
+	struct llist_node *freed_addr;
+
+	while ((freed_addr = llist_del_first(&linux_kmem_cache_list)) != NULL) {
+		struct cache_addr *caddr = (struct cache_addr *) freed_addr;
```

Comment:
freed_addr is a pointer to `struct llist_node`member rather than pointer to freed structure. Use `container_of` macro to convert one to another

Reply:
Thanks for telling me about `container_of`. I have incorporated it.